packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1014 @@
+            mergeType = MergeType.twoWay;
+          }
+          switch (mergeType) {
+            case MergeType.twoWay:
+              {
+                basePath = currentFile.path;
+                currentPath = currentFile.path;
+                targetPath = context.fileSystem.path.join(
+                    context
+                        .migrateResult.generatedTargetTemplateDirectory!.path,
+                    localPath);
+                break;
+              }
+            case MergeType.threeWay:
+              {
+                basePath = context.fileSystem.path.join(
+                    context.migrateResult.generatedBaseTemplateDirectory!.path,
+                    localPath);
+                currentPath = currentFile.path;
+                targetPath = context.fileSystem.path.join(
+                    context
+                        .migrateResult.generatedTargetTemplateDirectory!.path,
+                    localPath);
+                break;
+              }
+            case MergeType.custom:
+              {
+                break; // handled above
+              }
+          }
+          if (mergeType != MergeType.custom) {
+            result = await context.migrateUtils.gitMergeFile(
+              base: basePath,
+              current: currentPath,
+              target: targetPath,
+              localPath: localPath,
+            );
+          }
+        }
+        if (result != null) {
+          // Don't include if result is identical to the current file.
+          if (result is StringMergeResult) {
+            if (result.mergedString == currentFile.readAsStringSync()) {
+              context.status.pause();
+              context.logger
+                  .printStatus('$localPath was merged with a $mergeType.');
+              context.status.resume();
+              continue;
+            }
+          } else {
+            if ((result as BinaryMergeResult).mergedBytes ==
+                currentFile.readAsBytesSync()) {
+              continue;
+            }
+          }
+          context.migrateResult.mergeResults.add(result);
+        }
+        if (context.verbose) {
+          context.status.pause();
+          context.logger
+              .printStatus('$localPath was merged with a $mergeType.');
+          context.status.resume();
+        }
+        continue;
+      }
+    }
+
+    // Add files that are in the target, marked as always migrate, and missing in the current project.
+    for (final String localPath in missingAlwaysMigrateFiles) {
+      final File targetTemplateFile = context
+          .migrateResult.generatedTargetTemplateDirectory!
+          .childFile(localPath);
+      if (targetTemplateFile.existsSync() &&
+          !_skipped(localPath, context.fileSystem,
+              blacklistPrefixes: context.blacklistPrefixes)) {
+        context.migrateResult.addedFiles
+            .add(FilePendingMigration(localPath, targetTemplateFile));
+      }
+    }
+  }
+}
```

Comment:
Here too, consider extracting small helpers to allow someone to easily see the overall logical structure of the flow.